packages/telemetry/src/events/session.ts

```diff
@@ -0,0 +1,74 @@
+import escalade from 'escalade/sync';
+import { createRequire } from 'node:module';
+import { fileURLToPath } from 'node:url';
+
+const require = createRequire(import.meta.url);
+
+const EVENT_SESSION = 'ASTRO_CLI_SESSION_STARTED';
+
+interface EventCliSession {
+	astroVersion: string;
+	cliCommand: string;
+}
+
+interface ConfigInfo {
+	hasViteConfig: boolean;
```

Comment:
Talked in person, but you could remove `hasViteConfig`, `hasBase`, and `experimentalFeatures` if you instead created a new property called `configKeys` which just mapped over all used keys, to create an array of strings like this: `["site", "base", "build.format", "markdown.remarkPlugins"]`

Reply:
If we did it this way I don't think we could easily answer the question "What % is using the base config". Because there isn't a way to get a report on a string not being in an array.

Strings in an array is something that you can use to compare though, so if you wanted to compare `site` vs `base` that would be useful. But I don't know if that's what we're going for here.